pkg/console/controllers/service/controller.go

```diff
@@ -108,23 +110,55 @@ func (c *ServiceSyncController) sync() error {
 		return nil
 	case operatorsv1.Removed:
 		klog.V(4).Infoln("console is in a removed state: deleting service")
-		return c.removeService()
+		if err = c.removeService(api.OpenshiftConsoleRedirectServiceName); err != nil {
+			return err
+		}
+		return c.removeService(c.serviceName)
 	default:
 		return fmt.Errorf("unknown state: %v", operatorConfig.Spec.ManagementState)
 	}
 
 	updatedOperatorConfig := operatorConfig.DeepCopy()
-	_, _, svcErr := resourceapply.ApplyService(c.serviceClient, c.recorder, service.DefaultService(nil))
+
+	requiredSvc := service.DefaultService(updatedOperatorConfig)
+	_, _, svcErr := resourceapply.ApplyService(c.serviceClient, c.recorder, requiredSvc)
 	status.HandleProgressingOrDegraded(updatedOperatorConfig, "ServiceSync", "FailedApply", svcErr)
+	if svcErr != nil {
+		return svcErr
+	}
+
+	redirectSvcErrReason, redirectSvcErr := c.SyncRedirectService(updatedOperatorConfig)
```

Comment:
Have to create a new service for redirecting since the `Protocol` field on the Service is feature gated for 1.18. For that reason we cant really on updating it and we need to create a new service for redirecting